### ISSUE TITLE
Suboptimal placement of "missing with-clauses" error

### ISSUE THREAD

**Issue by guillaumebrunerie** [agda/agda#3262]
Consider the following code
```agda
data T : Set where
  a b c : T

f : T → Set
f a = ?
f b = ?
f c with c | c
... | _ = ?
```
(note that I am doing a `with` with two terms, but on the last line I only match with one of them)
 
The error `Missing with-clauses for function f` points to the `f : T → Set` line. It would be more helpful if it pointed to one of the last two lines. If the definition of `f` is big and has many patterns before the offending `with`, it can be hard to find where the problem is.

**Comment by andreasabel** [comment]
I'll attach the error location to the lhs with the with.

**Comment by andreasabel** [comment]
Fixed in eaa14b46a8605bfaaff6937025e3bdd9e7a2c7 (where I put #3263 by accident).
